services/horizon: fix reingest duplicate errors on same ledger range

sreuland opened the pull request:
<!-- If you're making a doc PR or something tiny where the below is irrelevant, delete this
template and use a short description, but in your description aim to include both what the
change is, and why it is being made, with enough context for anyone to understand. -->

<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

include liquidity pool tables in the ingest state machine's history operation purge for a given ledger range

### Why

fix error during reingestion due to duplicate errors in liquidity pool table state

closes #4110 

### Known limitations

N/A

tamirms commented:
@sreuland can you rebase on top of release-horizon-v2.12.0 and discard the commits which are not relevant?

sreuland commented:
> @sreuland can you rebase on top of release-horizon-v2.12.0 and discard the commits which are not relevant?

@tamirms , how about I squash merge this down to one commit to release-horizon-v2.12.0, then I can port that commit to master afterwards?

sreuland commented on services/horizon/internal/integration/db_test.go:
```diff
@@ -103,8 +209,8 @@ func TestReingestDB(t *testing.T) {
 
-	// Reingest into the DB
 	tt.NoError(horizoncmd.RootCmd.Execute())
+	tt.NoError(horizoncmd.RootCmd.Execute(), "Repeat the same reingest range against db, should not have errors.")
```
this is the crux of change, verifies repeated ingest works over same range on same db.

sreuland commented:
> > @sreuland can you rebase on top of release-horizon-v2.12.0 and discard the commits which are not relevant?
> 
> @tamirms , how about I squash merge this down to one commit to release-horizon-v2.12.0, then I can port that commit to master afterwards?

@tamirms , I rebased to the release branch and squashed, should be finished in CI soon, thanks!

tamirms commented:
great, thanks! I will merge
